exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -4,166 +4,193 @@
+
+    private final String[] board = {POP, SCIENCE, SPORTS, ROCK, POP, SCIENCE, SPORTS, ROCK, POP, SCIENCE, SPORTS, ROCK};
+
+    private ArrayList players = new ArrayList();
+    private int[] playerPositions = new int[MAX_NUMBER_OF_PLAYERS];
+    private int[] playerPurses = new int[MAX_NUMBER_OF_PLAYERS];
+    private boolean[] playersInPenaltyBox = new boolean[MAX_NUMBER_OF_PLAYERS];
+
+    private LinkedList popQuestions = new LinkedList();
+    private LinkedList scienceQuestions = new LinkedList();
+    private LinkedList sportsQuestions = new LinkedList();
+    private LinkedList rockQuestions = new LinkedList();
+
+    private int currentPlayerID = PLAYER_ONE_ID;
+    private boolean isGettingOutOfPenaltyBox;
+
+    public Game() {
+        for (int index = 0; index < NUMBER_OF_QUESTIONS; index++) {
+            popQuestions.addLast("Pop Question " + index);
+            scienceQuestions.addLast(("Science Question " + index));
+            sportsQuestions.addLast(("Sports Question " + index));
+            rockQuestions.addLast(createRockQuestion(index));
+        }
+    }
+
+    private String createRockQuestion(int index) {
+        return "Rock Question " + index;
+    }
+
+    public boolean addPlayer(String playerName) {
+        players.add(playerName);
+        playerPositions[getNumberOfPlayers()] = 0;
+        playerPurses[getNumberOfPlayers()] = 0;
+        playersInPenaltyBox[getNumberOfPlayers()] = false;
+
+        System.out.println(playerName + " was added");
+        System.out.println("They are player number " + players.size());
+        return true;
+    }
+
+    private int getNumberOfPlayers() {
+        return players.size();
+    }
+
+    public void processRoll(int rolledNumber) {
+        System.out.println(players.get(currentPlayerID) + " is the current player");
+        System.out.println("They have rolled a " + rolledNumber);
+
+        if (isPlayerInPenaltyBox(currentPlayerID)) {
+            if (isOddNumber(rolledNumber)) {
+                isGettingOutOfPenaltyBox = true;
+
+                System.out.println(players.get(currentPlayerID) + " is getting out of the penalty box");
+                movePlayer(rolledNumber, currentPlayerID);
+
+                System.out.println("The category is " + getCategoryForPosition());
+                askQuestion();
+            } else {
+                System.out.println(players.get(currentPlayerID) + " is not getting out of the penalty box");
+                isGettingOutOfPenaltyBox = false;
+            }
+
+        } else {
+            movePlayer(rolledNumber, currentPlayerID);
+            System.out.println("The category is " + getCategoryForPosition());
+            askQuestion();
+        }
+
+    }
+
+    private void printPlayerPosition(int playerID) {
+        System.out.println(players.get(playerID)
+                + "'s new location is "
+                + getPlayerPosition(playerID));
+    }
+
+    private boolean isOddNumber(int number) {

```

Comment:
I like this! Solves the issue we had yesterday where people were confused as to what the actual result was, though perhaps you could have kept the "rolledNumber" variable to continue being explicit